app/ide-desktop/lib/dashboard/src/tailwind.css

```diff
@@ -0,0 +1,226 @@
+            normal
+        );
+    }
+
+    body {
+        margin: 0; /* 1 */
+    }
+    
+    #dashboard {
+        line-height: inherit; /* 2 */
+    }
+
+    #dashboard hr {
+        height: 0; /* 1 */
+        color: inherit; /* 2 */
+        border-top-width: 1px; /* 3 */
+    }
+
+    #dashboard abbr:where([title]) {
+        text-decoration: underline dotted;
+    }
+
+    #dashboard h1, #dashboard h2, #dashboard h3, #dashboard h4, #dashboard h5, #dashboard h6 {
+        font-size: inherit;
+        font-weight: inherit;
+    }
+
+
+    #dashboard a {
+        color: inherit;
+        text-decoration: inherit;
+    }
+
+    #dashboard b, #dashboard strong {
+        font-weight: bolder;
+    }
+
+    #dashboard code, #dashboard kbd, #dashboard samp, #dashboard pre {
+        font-family: theme(
+            "fontFamily.mono",
+            ui-monospace,
+            SFMono-Regular,
+            Menlo,
+            Monaco,
+            Consolas,
+            "Liberation Mono",
+            "Courier New",
+            monospace
+        ); /* 1 */
+        font-size: 1em; /* 2 */
+    }
+
+    #dashboard small {
+        font-size: 80%;
+    }
+
+    #dashboard sub, #dashboard sup {
+        font-size: 75%;
+        line-height: 0;
+        position: relative;
+        vertical-align: baseline;
+    }
+
+    #dashboard sub {
+        bottom: -0.25em;
+    }
+
+    #dashboard sup {
+        top: -0.5em;
+    }
+
+    #dashboard table {
+        text-indent: 0; /* 1 */
+        border-color: inherit; /* 2 */
+        border-collapse: collapse; /* 3 */
+    }
+
+    #dashboard button, #dashboard input, #dashboard optgroup, #dashboard select, #dashboard textarea {
+        font-family: inherit; /* 1 */
+        font-size: 100%; /* 1 */
+        font-weight: inherit; /* 1 */
+        line-height: inherit; /* 1 */
+        color: inherit; /* 1 */
+        margin: 0; /* 2 */
+        padding: 0; /* 3 */
+    }
+
+    #dashboard button, #dashboard select {
+        text-transform: none;
+    }
+
+    #dashboard button, #dashboard [type="button"], #dashboard [type="reset"], #dashboard [type="submit"] {
+        -webkit-appearance: button;
+        background-color: transparent;
+        background-image: none;
+    }
+
+    #dashboard :-moz-focusring {
+        outline: auto;
+    }
+
+    #dashboard :-moz-ui-invalid {
+        box-shadow: none;
+    }
+
+    #dashboard progress {
+        vertical-align: baseline;
+    }
+
+    #dashboard ::-webkit-inner-spin-button, #dashboard ::-webkit-outer-spin-button {
+        height: auto;
+    }
+
+    #dashboard [type="search"] {
+        -webkit-appearance: textfield;
+        outline-offset: -2px;
+    }
+
+    #dashboard ::-webkit-search-decoration {
+        -webkit-appearance: none;
+    }
+
+    #dashboard ::-webkit-file-upload-button {
+        -webkit-appearance: button;
+        font: inherit;
+    }
+
+    #dashboard summary {
+        display: list-item;
+    }
```

Comment:
this looks like a generated file, it should not be committed.

Reply:
not a generated file, but rather manually created from tailwind's preflight. unfortunately there are no good solutions using purely tailwind. i can investigate whether this is possible using postcss + autoprefixer which is the normal solution

the reason this is required is that they all need the `#dashboard` prefix (*except* the one rule for `body`), otherwise the styles affect the IDE, causing things like #5963.

and they can't be left out either, otherwise the dashboard looks wrong

Reply:
(edit: `autoprefixer` is the wrong tool, and i believe `postcss-nested` (suggested by SO) *may* work but will cause tailwind to emit warnings)